Class forum component on homepage

zhushelly opened the pull request:
- Moved login/signup to the header bar (navbar)
- Created a class sidebar component that shows the user's currently joined classes
- Created a class forum component that allows users to create posts under a certain class
- Created server actions for post creation
- Added API routes for fetching posts by class ID
- Configured firebase-admin
- Configured session-based auth (cookies) 

To test:
- Join a class from /courses
- Go to home page and check if the class if shown in the sidebar
- Click on the class, the forum for that class will show up
- Create a post and check if it's rendered in the forum

Next steps:
- Create page for each post
- Server actions to update/delete post

Closes #37

zhushelly committed:
move login/signup to navbar

zhushelly committed:
added class forum page and server actions

zhushelly committed:
fix gray text

ZhenyuYu1 commented:
<img width="957" alt="Screenshot 2025-02-07 at 7 06 35 PM" src="https://github.com/user-attachments/assets/9048d571-730b-4665-9c53-f701f207308f" />

I'm getting this error when I login the page. I'm assuming that "joinedClasses" is new and if we previously created an account it didn't exist which is why its null. 
The error pops up as soon as I visit the homepage and when I try to select a course in /courses.

jeffsmithepic commented:
yeah I had the explore classes page add a new joinedClasses list to a user data if it didn't exist and then add a class ID if the user clicks the join class button, but the list may not exist for older accounts

So we might have to add something that automatically adds an empty joinedClasses list for accounts that don't have the list yet upon login or something

ZhenyuYu1 commented:
I think part of the problem should be temporary as it stems from "joinedClasses" being a new column added to each user's data so when we were testing it before it didn't exist so it gives an error. I fixed this by just saving my profile again which added in the column. It shouldn't be an issue for new users.

Another issue is in the profile page, when updating the profile, it should not set the joinedClasses to be an empty array each time the user saves their profile since this will delete their joinedClasses from before each time. I tested this by adding a class, then updating my profile, then seeing that the previously added class was no longer added.

AllenHsm committed:
Read the user data before setting profile to avoid overwriting the user's joinedClasses list

On the profile page, first read the user data to check whether it has an existing joinedClasses list. If not, then initialize it.

ZhenyuYu1 commented:
LGTM

jeffsmithepic commented:
lgtm

jeffsmithepic reviewed (approved):
lgtm
